core/templates/pages/exploration-player-page/services/current-interaction.service.spec.ts

```diff
@@ -16,134 +16,100 @@
+  let DUMMY_ANSWER = 'dummy_answer';
 
   // This mock is required since ContextService is used in
   // CurrentInteractionService to obtain the explorationId. So, in the
   // tests also we need to create a mock environment of exploration editor
   // since ContextService will error if it is used outside the context
   // of an exploration.
-  beforeEach(function() {
-    angular.mock.module(function($provide) {
-      $provide.value('UrlService', {
-        getPathname: function() {
-          return '/explore/123';
-        }
-      });
-      $provide.value('AngularNameService', new AngularNameService());
-      $provide.value('EditabilityService', new EditabilityService());
-      $provide.value('SolutionValidityService', new SolutionValidityService());
-      $provide.value(
-        'StateEditorService', new StateEditorService(
-          new SolutionValidityService()));
-      $provide.value('SuggestionModalService', new SuggestionModalService());
-      $provide.value('VoiceoverObjectFactory', new VoiceoverObjectFactory());
+  beforeEach(() => {
+    urlService = TestBed.get(UrlService);
+    spyOn(urlService, 'getPathname').and.callFake(() => {
+      return '/explore/123';
     });
+    currentInteractionService = TestBed.get(CurrentInteractionService);
   });
 
-  var DUMMY_ANSWER = 'dummy_answer';
-
-  var CurrentInteractionService;
-  beforeEach(angular.mock.inject(function($injector) {
-    CurrentInteractionService = $injector.get('CurrentInteractionService');
-  }));
 
-  it('should properly register onSubmitFn and submitAnswerFn', function() {
-    var answerState = null;
-    var dummyOnSubmitFn = function(answer, interactionRulesService) {
+  it('should properly register onSubmitFn and submitAnswerFn', () => {
+    let answerState = null;
+    let dummyOnSubmitFn = function(answer, interactionRulesService) {
       answerState = answer;
     };
 
-    CurrentInteractionService.setOnSubmitFn(dummyOnSubmitFn);
-    CurrentInteractionService.onSubmit(DUMMY_ANSWER, null);
+    currentInteractionService.setOnSubmitFn(dummyOnSubmitFn);
+    currentInteractionService.onSubmit(DUMMY_ANSWER, null);
     expect(answerState).toEqual(DUMMY_ANSWER);
 
     answerState = null;
-    var dummySubmitAnswerFn = function() {
-      CurrentInteractionService.onSubmit(DUMMY_ANSWER, null);
+    let dummySubmitAnswerFn = function() {
+      currentInteractionService.onSubmit(DUMMY_ANSWER, null);
     };
-    CurrentInteractionService.registerCurrentInteraction(
+    currentInteractionService.registerCurrentInteraction(
       dummySubmitAnswerFn, null);
-    CurrentInteractionService.submitAnswer();
+    currentInteractionService.submitAnswer();
     expect(answerState).toEqual(DUMMY_ANSWER);
   });
 
-  it('should properly register validityCheckFn', function() {
-    var dummyValidityCheckFn = function() {
+  it('should properly register validityCheckFn', () => {
+    let dummyValidityCheckFn = function() {
```

Comment:
Can we please change 'function' to fat arrow functions `() =>` here and everywhere?